### ISSUE TITLE
How do I use basic auth but only for certain routes?

### ISSUE THREAD

**Issue by lovedoodle** [LionC/express-basic-auth#14]
I have a couple public routes where I don't want there to be any baisc auth requirements.
Does this package offer a way to do that?

**Comment by kitajchuk** [comment]
@LionC I'm really interested in doing this as well. Can the module do this currently?

**Comment by johnmuhl** [comment]
Looks like it is planned for v2. See #13.

**Comment by num-lock** [comment]
Can't you simply use an express router and only _use_ it in there or only specify it for your desired routes? I mean this is a standard express middleware after all. At least that's how I am doing it. Just don't use it globally with `app.use()` if you don't want it globally.

**Comment by Murmand** [comment]
As @num-lock wrote, just use

```javascript
app.get('/route', basicAuth(...), (req, res) => {...});
```

instead of

```javascript
app.use(basicAuth(...));
```

**Comment by num-lock** [comment]
Yes, or even better: Use a `Router`. That's what they are there for after all. Putting everything directly into `app` is bad practice anyways.

```js
const router = express.Router();
router.get('/a', (req, res) => {...});
router.get('/b', (req, res) => {...});
app.use('/api', basicAuth(...), router);
// /api/a and /api/b now use basic-auth middleware
```

Alternatively you can also do `router.use(basicAuth(...))` first instead of putting it into `app.use` when applying the route. Either way it's a lot more concise and transparent than having the middleware itself handle which routes to hook on.

**Comment by horak** [comment]
Related question:

Can basicAuth challenge a single user on multiple routes? 

```javascript
/** everyone enters this password **/
app.get('/', basicAuth({
  users: { preview: 'pass123' },
  challenge: true
}));

/** hidden route for internal use **/
app.get('internal/', basicAuth({
  users: { internal: 'passabc' },
  challenge: true
}));
```

I was thinking maybe I'd need to have two different instances of basicAuth, but that didn't work either:

```javascript
const basicAuthPreview  = require('express-basic-auth');
const basicAuthInternal  = require('express-basic-auth');
```
